packages/curvenote-cli/src/plugins/articles.ts

```diff
@@ -0,0 +1,75 @@
+import type { DirectiveSpec, GenericNode } from 'myst-common';
+
+export const articlesDirective: DirectiveSpec = {
+  name: 'cn:articles',
+  doc: 'A listing directive that can be used to show a list of articles from a specific venue, collection or kind.',
+  arg: {
+    type: 'myst',
+    doc: 'The title of the listing.',
+  },
+  body: {
+    type: 'myst',
+    doc: 'Descriptive content to be displayed along with the listing.',
+  },
+  options: {
+    venue: {
+      type: String,
+      doc: 'The venue to list articles from.',
+      required: true,
+    },
+    collection: {
+      type: String,
+      doc: 'The collection to list articles from.',
+      required: false,
+    },
+    status: {
+      type: String,
+      doc: 'The status of articles to list (published | in-review).',
+      required: false,
+    },
+    display: {
+      type: String,
+      doc: 'The style of listing to display (list | cards).',
+      required: false,
+    },
+    limit: {
+      type: Number,
+      doc: 'The maximum number of articles to list.',
+      required: false,
+    },
+    pagination: {
+      type: String,
+      doc: 'If `limit` is set, this gives a hint to the renderer on how to handle pagination (more | all | infinite).',
```

Comment:
🤔 We could extend these directive spec options to use the same `options` validation we are using for template / check options... that would give us access to more declarative validation like `choices`...

Just musing - no influence on this PR 🙂